### DISCUSSION TITLE
Calling CreateServer (/next/dist/server/next.d.ts)  cannot return NextServer object after 13.4.3

### DISCUSSION THREAD

**Discussion by raychunghk** [vercel/next.js#52886] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Next.js 13.4.3 and newer
Nest.js 10
```

### Which example does this report relate to?

n/a

### Describe the Bug

My project is setup with nest.js 10, next.js, and bridged with nest-next.
In this function imports
```
    RenderModule.forRootAsync(
       Next({ dev: true }),
      /* null means that nest-next 
                should look for pages in root dir */
      { passthrough404: true, viewsDir: null },
    
    ),
```
inside below, The Next({ dev: true }) param is good because it is able to return a NextServer object in the file:
/node_modules/next/dist/server/next.d.ts
But this is not after version next 13.4.3
```
imports: [
    RenderModule.forRootAsync(
       Next({ dev: true }),
      /* null means that nest-next 
                should look for pages in root dir */
      { passthrough404: true, viewsDir: null },
    
    ),
    StaffModule,
    TestModule,
    TimesheetModule,
    LeaveRequestModule,
    VacationsModule,
    AuthModule,
    TypeGraphQLModule.forRoot({
      driver: ApolloDriver,
      emitSchemaFile: true,
      validate: false,
      path: '/absproxy/5000/graphql'
      , authChecker: ({ context }, roles) => {
        return true;
      },
  
    }), RecipeModule,
  ],
  controllers: [AppController],

  providers: [AppService, UserResolver, PrismaService],
})
export class AppModule { }

```

### Expected Behavior

The CreateServer function inside  should return a NextServer object.
/config/workspace/vm/js/NxTime/node_modules/next/dist/server/next.d.ts

### To Reproduce

Create a empty nest project
```
$ npm i -g @nestjs/cli
$ nest new project-name
```
add Next.js 13.4.2 in the project
```
 npm install next react react-dom
```

In the main.ts, import Next and log the object by calling "Next" like below
```
import Next from 'next';
async function bootstrap() {
 
  const app =   Next({ dev: true })
  app.prepare();
  console.log(app)
}
```
With Next version:  13.4.2 or lower, the above log will report
```
NextServer {
  options: { dev: true },
  serverPromise: Promise {
    <pending>,
    [Symbol(async_id_symbol)]: 41,
    [Symbol(trigger_async_id_symbol)]: 37
  }
}
```
With Next version >=13.4.3 
the app object is logged as {}

**Comment by tunatoksoz** [comment]
I believe this is the place things changed
https://github.com/vercel/next.js/blob/5e36c349925ddf71ab48ef44e3d713ddc5afc883/packages/next/src/server/next.ts#L281


Presumably changed after this difF:
https://github.com/vercel/next.js/commit/c7e5d767177b862506648fa87d96e80b7c358719

**Comment by lluiscab** [comment]
Taking a look at the current status of this issue, it all seems to be related to whatever or not the `customServer` property is set when initializing a `NextServer` instance. When this value is set to `true` (or more specifically, not explicitly set to `false`), then instead of getting an instance of `NextServer` we get a Proxy that limits the access we have to `NextServer` properties.

The following code change fixes this issue:

```diff
diff --git a/packages/next/src/server/next.ts b/packages/next/src/server/next.ts
index fabcea9..b6416ef 100644
--- a/packages/next/src/server/next.ts
+++ b/packages/next/src/server/next.ts
@@ -413,10 +413,16 @@ function createServer(options: NextServerOptions): NextServer {
               }
             }
             default: {
-              const method = server[propKey as keyof NextServer]
-              if (typeof method === 'function') {
-                return method.bind(server)
+              const methodOrProperty = server[propKey as keyof NextServer]
+              if (typeof methodOrProperty === 'function') {
+                return methodOrProperty.bind(server)
               }

+              return methodOrProperty
             }
           }
         },
```

Unfortunately, this change isn't enough, the proxy overrides the `prepare()` function and it's current implementation does not set the properties we're interested on (`app.server` (see https://github.com/kyle-mccarthy/nest-next/blob/master/lib/render.module.ts#L31) & `app.dynamicRoutes` (see https://github.com/kyle-mccarthy/nest-next/blob/master/lib/render.module.ts#L37) ).

Could we get some input from a project maintainer such as @shuding or @ijjk? (both have recently modified this file, and @shuding introduced the proxy return)

**Reply by razyalov** [reply]
Is there any movement on this issue? Did anyone find a solution to this problem?

**Comment by sergeysportcore** [comment]
Any updates?

**Comment by kaminion** [comment]
It's going on, it's forcing to use low version

**Comment by Amantel** [comment]
Any updates?

**Comment by raychunghk** [comment]
I give up this approach.
I change my app to make next.js as reverseproxy to make api routes rewritten to the nest.js backend instead.
It is leaner solution compared to nest-next, and that project is no longer updated.
![8M9YCywfTI](https://github.com/user-attachments/assets/a7bdc086-9804-41fe-9fe9-d2e2408bdf6c)
